authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/client/ClientPolicyProvider.java

```diff
@@ -0,0 +1,37 @@
+package org.keycloak.authorization.policy.provider.client;
+
+import org.keycloak.authorization.model.Policy;
+import org.keycloak.authorization.policy.evaluation.Evaluation;
+import org.keycloak.authorization.policy.evaluation.EvaluationContext;
+import org.keycloak.authorization.policy.provider.PolicyProvider;
+
+import static org.keycloak.authorization.policy.provider.client.ClientPolicyProviderFactory.getClients;
+
+public class ClientPolicyProvider implements PolicyProvider {
+
+    private final Policy policy;
+
+    public ClientPolicyProvider(Policy policy) {
+        this.policy = policy;
+    }
+
+    @Override
+    public void evaluate(Evaluation evaluation) {
+        EvaluationContext context = evaluation.getContext();
+        String[] clientIds = getClients(this.policy);
+
+        if (clientIds.length > 0) {
+            for (String clientId : clientIds) {
+                if (context.getIdentity().getId().equals(clientId)) {
```

Comment:
```Identity.getId``` does not always give you the id of the client, but only when the token was issued to the resource server itself (e.g.: when obtaining a PAT to access the Protection API).

To achieve what you want, you should use the ```aud``` claim within the token, which provides the clients to which the token was issued.